File: users_microservice/src/main/java/com/zigzag/crm/framework/graphql/features/user/mutation/CrmUserMutation.java
package com.zigzag.crm.framework.graphql.features.user.mutation;

import com.zigzag.crm.usecase.feature.user.api.Usecase_CreateUser;
import com.zigzag.crm.usecase.feature.user.dto.CrmUserDto;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.MutationMapping;
import org.springframework.stereotype.Controller;
import reactor.core.publisher.Mono;

@Controller
@RequiredArgsConstructor
public class CrmUserMutation {

    @Autowired
    private Usecase_CreateUser usecaseCreateUser;

    @MutationMapping
    public Mono<CrmUserDto.Response.Public> createUser( @Argument CrmUserDto.Request.Create user)  {
        return usecaseCreateUser.execute(user);
    }
}

File: framework/graphql/src/main/com/zigzag/crm/framework/graphql/mutation/CrmUserMutation.kt
package com.zigzag.crm.module.graphql.mutation

import com.zigzag.crm.framework.domain.api.features.user.CrmUser
import com.zigzag.crm.framework.domain.api.repository.ICrmUserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Controller
class CrmUserMutation {
    @Autowired
    private lateinit var userRepository: ICrmUserRepository;

    fun newUser(user: CrmUserDto): Mono<CrmUser> {
        return userRepository.createUser(user);
    }
}
